### PR TITLE
Finalization for custom gates (reborn)

### PR DESCRIPTION
**Pull request by rbonichon**
Re-opening https://github.com/MinaProtocol/mina/pull/12382 after manual revert

### CODE REVIEW

File: scripts/testone.sh [hunk 1]
```diff
@@ -27,7 +27,9 @@ if [[ "$#" -eq "1" ]]; then
 else 
     TEST_CASE="$TEST_FILE:$2"
 fi
-ulimit -s 65532 && (ulimit -n 10240 || true) && \
+
+( ulimit -s 65532 || true )
```
Comment (mrmr1993) [inline scripts/testone.sh:31]: We should put back the old `ulimit -n`

File: src/lib/pickles/wrap.ml [hunk 2]
```diff
@@ -283,98 +283,352 @@ let deferred_values (type n) ~(sgs : (Backend.Tick.Curve.Affine.t, n) Vector.t)
...
+              ~feature_flags
+              ~step_domains:
+                (`Known
+                  [ { h = Pow_2_roots_of_unity vk.domain.log_size_of_group } ]
+                  )
+              ~sponge ~prev_challenges:[] deferred_values evals
+          in
+
+          (* Actually do the above as the prover *)
+          Impls.Step.(As_prover.(fun () -> read Boolean.typ res)) )
```
Comment (mrmr1993) [inline src/lib/pickles/wrap.ml:475]: Nit: `read the result boolean from the circuit and make it available to the OCaml world`

**Review by mrmr1993 — APPROVED** [review]

File: src/lib/pickles/wrap.ml (hunk 2 above)
Comment (rbonichon) [reply src/lib/pickles/wrap.ml:475]: Isn't that done by:

Comment (rbonichon) [reply src/lib/pickles/wrap.ml:476]: Got it 🙏

**Review by mrmr1993 — APPROVED** [review]

**Pull request merged**